spec/js_routes/options_spec.rb

```diff
@@ -404,6 +398,42 @@
         end
       end
     end
+
+    context 'when window.location is present' do
+      let(:current_protocol) { 'http:' } # window.location.protocol includes the colon character
+      let(:current_hostname) { 'current.example.com' }
+      let(:current_port){ '' } # an empty string means port 80
+      let(:current_host) do
+        host = "#{current_hostname}"
+        host += ":#{current_port}" unless current_port == ''
+        host
+      end
+
+      before do
+        jscontext['window'] = {
+          :location => {
+            :protocol => current_protocol,
+            :hostname => current_hostname,
+            :port => current_port,
+            :host => current_host
+          }
+        }
+      end
+
+      context "without specifying a default host" do
+        let(:_options) { { :url_links => true } }
+
+        it "uses the current host" do
+          expect { JsRoutes.generate(_options) }.not_to raise_error
+          expect(evaljs("Routes.inbox_path")).not_to be_nil
+          expect(evaljs("Routes.inbox_url")).not_to be_nil
```

Comment:
I don't know why these 3 assertions above make sense. Are we just fine having ones below?